### PR TITLE
Rework implementation of the Properties admin facet in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This tiny PR mostly changes `INativePropertyAdmin`, which is now using an event.

### CODE REVIEW

File: csharp/src/Ice/PropertiesAdmin.cs [hunk 1]
```diff
@@ -7,33 +7,31 @@
...
+        private readonly Communicator _communicator;
+        private readonly ILogger _logger;
 
-        public Dictionary<string, string>
-        GetPropertiesForPrefix(string name, Ice.Current current) => _communicator.GetProperties(forPrefix: name);
+        public string GetProperty(string name, Current current) => _communicator.GetProperty(name) ?? "";
 
-        public void
-        SetProperties(Dictionary<string, string> props, Ice.Current current)
+        public Dictionary<string, string> GetPropertiesForPrefix(string name, Current current) =>
```
Comment (pepone) [inline csharp/src/Ice/PropertiesAdmin.cs:31]: I will use `prefix` for the argument here instead of `name`

**Comment by bernardnormier** [comment]
I've just updated it further and replaced `INativePropertiesAdmin` by a partial interface `IPropertiesAdmin`. This way, no need for this "native" interface. I also added `partial` to the generation of interface proxies and skeletons.

Also synchronized the parameter names in PropertiesAdmin with those used in the Slice definition.

File: csharp/src/Ice/PropertiesAdmin.cs [hunk 2]
```diff
@@ -7,9 +7,9 @@
 
 namespace Ice
 {
-    /// <summary>Implementations of the Properties admin facet implement both IPropertiesAdmin (which corresponds
-    /// to the Slice interface PropertiesAdmin) and the local INativePropertiesAdmin interface.</summary>
-    public interface INativePropertiesAdmin
+    /// <summary>This partial interface extends the IPropertieAdmin interface generated from the Slice interface
+    /// PropertiesAdmin. All Properties admin facet implement this interface.</summary>
```
Comment (pepone) [inline csharp/src/Ice/PropertiesAdmin.cs:11]: `All Properties admin facet implement this interface`
I don't understand this sentence, what do you mean?

Comment (bernardnormier) [reply csharp/src/Ice/PropertiesAdmin.cs:11]: I removed it as it's indeed unclear here.

**Pull request merged**